### PR TITLE
[SPARK-33604][SQL] Group exception messages in sql/execution

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR group exception messages in `/core/src/main/scala/org/apache/spark/sql/execution`.

### Why are the changes needed?
It will largely help with standardization of error messages and its maintenance.

### Does this PR introduce _any_ user-facing change?
No. Error messages remain unchanged.

### How was this patch tested?
No new tests - pass all original tests to make sure it doesn't break any existing behavior.

### CODE REVIEW

**Comment by beliefer** [comment]
retest this please

**Comment by beliefer** [comment]
retest this please

**Comment by beliefer** [comment]
ping @allisonwang-db

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 1]
```diff
@@ -556,4 +559,179 @@ object QueryExecutionErrors {
...
+  def rowLargerThanLimitUnsupportedError(): Throwable = {
+    new UnsupportedOperationException("Does not support row that is larger than 256M")
+  }
+
+  def cannotBuildHashedRelationWithUniqueKeysExceededError(): Throwable = {
+    new UnsupportedOperationException(
+      "Cannot build HashedRelation with more than 1/3 billions unique keys")
+  }
+
+  def cannotBuildHashedRelationLargerThanLimitError(): Throwable = {
```
Comment (allisonwang-db) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:697]: cannotBuildHashedRelationLargerThan8GError

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 2]
```diff
@@ -556,4 +559,179 @@ object QueryExecutionErrors {
...
+  def unexpectedWindowFunctionFrameError(frame: String): Throwable = {
+    new RuntimeException(s"Unexpected window function frame $frame.")
+  }
+
+  def cannotParseStatisticAsPercentileError(
+      stats: String, e: NumberFormatException): Throwable = {
+    new IllegalArgumentException(s"Unable to parse $stats as a percentile", e)
+  }
+
+  def statisticIsNotRecognisedStatisticError(stats: String): Throwable = {
```
Comment (allisonwang-db) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:725]: statisticNotRecognizedError

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala [hunk 3]
```diff
@@ -303,4 +303,64 @@ object QueryParsingErrors {
     new ParseException(s"Found duplicate keys '$key'.", ctx)
   }
 
+  def formatForSetConfigurationUnExpectedError(ctx: SetConfigurationContext): Throwable = {
```
Comment (allisonwang-db) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:306]: unexpectedFomatForSetConfigurationError

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala [hunk 4]
```diff
@@ -303,4 +303,64 @@ object QueryParsingErrors {
...
+    new ParseException(
+      s"""
+         |Expected format is 'SET', 'SET key', or 'SET key=value'. If you want to include
+         |special characters in key, or include semicolon in value, please use quotes,
+         |e.g., SET `ke y`=`v;alue`.
+       """.stripMargin.replaceAll("\n", " "), ctx)
+  }
+
+  def invalidPropertyKeyForSetQuotedConfigurationError(
```
Comment (allisonwang-db) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:315]: invalidPropertyKeyForSetConfigurationError

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala [hunk 5]
```diff
@@ -303,4 +303,64 @@ object QueryParsingErrors {
...
+      keyCandidate: String, valueStr: String, ctx: SetQuotedConfigurationContext): Throwable = {
+    new ParseException(s"'$keyCandidate' is an invalid property key, please " +
+      s"use quotes, e.g. SET `$keyCandidate`=`$valueStr`", ctx)
+  }
+
+  def invalidPropertyValueForSetQuotedConfigurationError(
```
Comment (allisonwang-db) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:321]: invalidPropertyValueForSetConfigurationError

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala [hunk 6]
```diff
@@ -303,4 +303,64 @@ object QueryParsingErrors {
...
+      valueCandidate: String, keyStr: String, ctx: SetQuotedConfigurationContext): Throwable = {
+    new ParseException(s"'$valueCandidate' is an invalid property value, please " +
+      s"use quotes, e.g. SET `$keyStr`=`$valueCandidate`", ctx)
+  }
+
+  def formatForResetConfigurationUnExpectedError(ctx: ResetConfigurationContext): Throwable = {
```
Comment (allisonwang-db) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:327]: unexpectedFormatForResetConfigurationError

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala [hunk 7]
```diff
@@ -303,4 +303,64 @@ object QueryParsingErrors {
...
+  def rowFormatNotUsedWithStoredAsError(ctx: CreateTableLikeContext): Throwable = {
+    new ParseException("'ROW FORMAT' must be used with 'STORED AS'", ctx)
+  }
+
+  def useDefinedRecordReaderOrWriterClassesError(ctx: ParserRuleContext): Throwable = {
+    new ParseException(
+      "Unsupported operation: Used defined record reader/writer classes.", ctx)
+  }
+
+  def directoryPathAndOptionPathBothSpecifiedError(ctx: InsertOverwriteDirContext): Throwable = {
```
Comment (allisonwang-db) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:357]: directoryPathAndOptionsPathBothSpecifiedError

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala [hunk 8]
```diff
@@ -303,4 +303,64 @@ object QueryParsingErrors {
...
+    new ParseException(
+      "Directory path and 'path' in OPTIONS should be specified one, but not both", ctx)
+  }
+
+  def illegalLocalFileSchemaError(ctx: InsertOverwriteDirContext): Throwable = {
```
Comment (allisonwang-db) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:362]: unsupportedLocalFileSchemeError

**Review by allisonwang-db — COMMENTED** [review]
Looks good!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 9]
```diff
@@ -1087,4 +1087,42 @@ private[spark] object QueryCompilationErrors {
...
+         |which one. Please alias the Datasets with different names via `Dataset.as`
+         |before joining them, and specify the column using qualified name, e.g.
+         |`df.as("a").join(df.as("b"), $$"a.id" > $$"b.id")`. You can also set
+         |${SQLConf.FAIL_AMBIGUOUS_SELF_JOIN_ENABLED.key} to false to disable this check.
+       """.stripMargin.replaceAll("\n", " "))
+  }
+
+  def unexpectedEvalTypesForUDFsError(evalTypes: Set[Int]): Throwable = {
+    new AnalysisException(
+      s"Expected udfs have the same evalType but got different evalTypes: " +
```
Comment (wangyum) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:1121]: Do not need `s`.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala (hunk 4 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:315]: It seems we should keep `invalidPropertyKeyForSetQuotedConfigurationError`.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala (hunk 5 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:321]: It seems we should keep `invalidPropertyValueForSetQuotedConfigurationError`.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala (hunk 9 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:1332]: This PR just group exception messages. There need other PR to improve messages.

**Comment by beliefer** [comment]
retest this please

**Comment by beliefer** [comment]
ping @allisonwang-db

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala (hunk 5 above)
Comment (allisonwang-db) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:321]: Sounds good. We can emphasize the property key/value should be quoted.

**Comment by allisonwang-db** [comment]
Looks good to me!

**Comment by beliefer** [comment]
cc @cloud-fan

**Comment by beliefer** [comment]
cc @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ExplainUtils.scala [hunk 10]
```diff
@@ -205,7 +206,7 @@ object ExplainUtils extends AdaptiveSparkPlanHelper {
     case iter: Iterable[_] => s"${fieldName} [${iter.size}]: ${iter.mkString("[", ", ", "]")}"
     case str: String if (str == null || str.isEmpty) => s"${fieldName}: None"
     case str: String => s"${fieldName}: ${str}"
-    case _ => throw new IllegalArgumentException(s"Unsupported type for argument values: $values")
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ExplainUtils.scala:208]: isn't it assert like?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala [hunk 11]
```diff
@@ -51,7 +52,7 @@ object SQLExecution {
       // started execution of a query didn't call withNewExecutionId. The execution ID should be
       // set by calling withNewExecutionId in the action that begins execution, like
       // Dataset.collect or DataFrameWriter.insertInto.
-      throw new IllegalStateException("Execution ID should be set")
+      throw QueryExecutionErrors.executionIDNotSetError()
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala:55]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala [hunk 12]
```diff
@@ -174,7 +175,7 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
    */
   final def execute(): RDD[InternalRow] = executeQuery {
     if (isCanonicalizedPlan) {
-      throw new IllegalStateException("A canonicalized plan is not supposed to be executed.")
+      throw QueryExecutionErrors.executeCanonicalizedPlanError()
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala:178]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala [hunk 13]
```diff
@@ -187,7 +188,7 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
    */
   final def executeBroadcast[T](): broadcast.Broadcast[T] = executeQuery {
     if (isCanonicalizedPlan) {
-      throw new IllegalStateException("A canonicalized plan is not supposed to be executed.")
+      throw QueryExecutionErrors.executeCanonicalizedPlanError()
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala:191]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala [hunk 14]
```diff
@@ -201,7 +202,7 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
    */
   final def executeColumnar(): RDD[ColumnarBatch] = executeQuery {
     if (isCanonicalizedPlan) {
-      throw new IllegalStateException("A canonicalized plan is not supposed to be executed.")
+      throw QueryExecutionErrors.executeCanonicalizedPlanError()
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala:205]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala [hunk 15]
```diff
@@ -294,7 +295,7 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
    * Overridden by concrete implementations of SparkPlan.
    */
   protected[sql] def doExecuteBroadcast[T](): broadcast.Broadcast[T] = {
-    throw new UnsupportedOperationException(s"$nodeName does not implement doExecuteBroadcast")
+    throw QueryExecutionErrors.doExecuteBroadcastNotImplementedError(nodeName)
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala:298]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala [hunk 16]
```diff
@@ -303,8 +304,7 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
    * when it is no longer needed. This allows input formats to be able to reuse batches if needed.
    */
   protected def doExecuteColumnar(): RDD[ColumnarBatch] = {
-    throw new IllegalStateException(s"Internal Error ${this.getClass} has column support" +
-      s" mismatch:\n${this}")
+    throw QueryExecutionErrors.sparkPlanHasColumnSupportMismatchError(this.getClass, this.toString)
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala:307]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala [hunk 17]
```diff
@@ -618,29 +617,26 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
         LocalTableScanExec(output, sink.allData.map(r => toRow(r).copy())) :: Nil
 
       case logical.Distinct(child) =>
-        throw new IllegalStateException(
-          "logical distinct operator should have been replaced by aggregate in the optimizer")
+        throw QueryExecutionErrors.logicalOperatorNotReplacedError(
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala:620]: ditto

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala:620]: and all the changes in `BasicOperators`

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala [hunk 18]
```diff
@@ -65,7 +66,7 @@ case class CustomShuffleReaderExec private(
             case other => other
           }
         case _ =>
-          throw new IllegalStateException("operating on canonicalization plan")
+          throw QueryExecutionErrors.operatingOnCanonicalizationPlanError()
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala:69]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala [hunk 19]
```diff
@@ -108,7 +109,7 @@ case class CustomShuffleReaderExec private(
         case CoalescedPartitionSpec(startReducerIndex, endReducerIndex) =>
           startReducerIndex.until(endReducerIndex).map(bytesByPartitionId).sum
         case p: PartialReducerPartitionSpec => p.dataSize
-        case p => throw new IllegalStateException("unexpected " + p)
+        case p => throw QueryExecutionErrors.unexpectedShufflePartitionSpecError(p.toString)
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala:112]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala [hunk 20]
```diff
@@ -184,7 +185,7 @@ case class CustomShuffleReaderExec private(
         sendDriverMetrics()
         stage.shuffle.getShuffleRDD(partitionSpecs.toArray)
       case _ =>
-        throw new IllegalStateException("operating on canonicalized plan")
+        throw QueryExecutionErrors.operatingOnCanonicalizedPlanError()
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala:188]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala [hunk 21]
```diff
@@ -165,7 +165,7 @@ case class ShuffleQueryStageExec(
     case s: ShuffleExchangeLike => s
     case ReusedExchangeExec(_, s: ShuffleExchangeLike) => s
     case _ =>
-      throw new IllegalStateException("wrong plan for shuffle stage:\n " + plan.treeString)
+      throw QueryExecutionErrors.wrongPlanForShuffleStageError(plan.treeString)
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala:168]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala [hunk 22]
```diff
@@ -217,7 +217,7 @@ case class BroadcastQueryStageExec(
     case b: BroadcastExchangeLike => b
     case ReusedExchangeExec(_, b: BroadcastExchangeLike) => b
     case _ =>
-      throw new IllegalStateException("wrong plan for broadcast stage:\n " + plan.treeString)
+      throw QueryExecutionErrors.wrongPlanForBroadcastStageError(plan.treeString)
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala:220]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/RowBasedHashMapGenerator.scala [hunk 23]
```diff
@@ -129,7 +130,7 @@ class RowBasedHashMapGenerator(
         case t: DataType =>
           if (!t.isInstanceOf[StringType] && !t.isInstanceOf[CalendarIntervalType] &&
             !CodeGenerator.isPrimitiveType(t)) {
-            throw new IllegalArgumentException(s"cannot generate code for unsupported type: $t")
+            throw QueryExecutionErrors.cannotGenerateCodeForUnsupportedTypeError(t)
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/RowBasedHashMapGenerator.scala:133]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala [hunk 24]
```diff
@@ -456,8 +457,7 @@ class TungstenAggregationIterator(
       hashMap.free()
       resultCopy
     } else {
-      throw new IllegalStateException(
-        "This method should not be called when groupingExpressions is not empty.")
+      throw QueryExecutionErrors.callMethodWithEmptyGroupingExpressionsError()
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala:460]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala [hunk 25]
```diff
@@ -58,7 +59,7 @@ class DefaultCachedBatchSerializer extends SimpleMetricsCachedBatchSerializer {
       schema: Seq[Attribute],
       storageLevel: StorageLevel,
       conf: SQLConf): RDD[CachedBatch] =
-    throw new IllegalStateException("Columnar input is not supported")
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala:61]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala [hunk 26]
```diff
@@ -347,7 +348,7 @@ private[columnar] case object RunLengthEncoding extends CompressionScheme {
           decompress0(columnVector, capacity, getInt, putInt)
         case _: LongType =>
           decompress0(columnVector, capacity, getLong, putLong)
-        case _ => throw new IllegalStateException("Not supported type in RunLengthEncoding.")
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala:350]: ditto

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala:350]: and other two in this file

File: sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala [hunk 27]
```diff
@@ -359,8 +355,7 @@ trait HashJoin extends JoinCodegenSupport {
       case LeftAnti => codegenAnti(ctx, input)
       case _: ExistenceJoin => codegenExistence(ctx, input)
       case x =>
-        throw new IllegalArgumentException(
-          s"HashJoin should not take $x as the JoinType")
+        throw QueryExecutionErrors.hashJoinCannotTakeJoinTypeError(x)
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala:358]: ditto in this file

File: sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala [hunk 28]
```diff
@@ -53,8 +54,8 @@ class SQLMetric(val metricType: String, initValue: Long = 0L) extends Accumulato
     case o: SQLMetric =>
       if (_value < 0) _value = 0
       if (o.value > 0) _value += o.value
-    case _ => throw new UnsupportedOperationException(
-      s"Cannot merge ${this.getClass.getName} with ${other.getClass.getName}")
+    case _ => throw QueryExecutionErrors.cannotMergeSQLMetricWithAccumulatorV2Error(
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala:57]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala [hunk 29]
```diff
@@ -200,7 +201,7 @@ object SQLMetrics {
       } else if (metricsType == NS_TIMING_METRIC) {
         duration => Utils.msDurationToString(duration.nanos.toMillis)
       } else {
-        throw new IllegalStateException("unexpected metrics type: " + metricsType)
+        throw QueryExecutionErrors.unexpectedMetricsTypeError(metricsType)
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/metric/SQLMetrics.scala:204]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/python/ExtractPythonUDFs.scala [hunk 30]
```diff
@@ -268,7 +266,7 @@ object ExtractPythonUDFs extends Rule[LogicalPlan] with PredicateHelper {
             case PythonEvalType.SQL_SCALAR_PANDAS_UDF | PythonEvalType.SQL_SCALAR_PANDAS_ITER_UDF =>
               ArrowEvalPython(validUdfs, resultAttrs, child, evalType)
             case _ =>
-              throw new AnalysisException("Unexpected UDF evalType")
+              throw QueryCompilationErrors.unexpectedUDFEvalTypeError()
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/python/ExtractPythonUDFs.scala:269]: ditto, we can even change to `IllegalStateException`

**Review by cloud-fan — APPROVED** [review]
LGTM except for some assert-like errors.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ExplainUtils.scala (hunk 10 above)
Comment (beliefer) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/ExplainUtils.scala:208]: OK

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala (hunk 17 above)
Comment (beliefer) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala:620]: OK

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala (hunk 26 above)
Comment (beliefer) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala:350]: OK

File: sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala [hunk 31]
```diff
@@ -456,7 +456,7 @@ class TungstenAggregationIterator(
       hashMap.free()
       resultCopy
     } else {
-      throw new IllegalStateException(
+      throw  new IllegalStateException(
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala:459]: nit: extra space

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala [hunk 32]
```diff
@@ -347,7 +348,7 @@ private[columnar] case object RunLengthEncoding extends CompressionScheme {
...
+        case _ => throw new IllegalStateException(s"Not supported type in RunLengthEncoding.")
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala:351]: nit: `s"..."` s is not needed

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala [hunk 33]
```diff
@@ -517,7 +517,7 @@ private[columnar] case object DictionaryEncoding extends CompressionScheme {
             }
             pos += 1
           }
-        case _ => throw new IllegalStateException("Not supported type in DictionaryEncoding.")
+        case _ => throw new IllegalStateException(s"Not supported type in DictionaryEncoding.")
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/compression/compressionSchemes.scala:520]: ditto

File: sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala [hunk 34]
```diff
@@ -69,16 +70,14 @@ trait HashJoin extends JoinCodegenSupport {
       joinType match {
         case _: InnerLike | RightOuter => right.outputPartitioning
         case x =>
-          throw new IllegalArgumentException(
-            s"HashJoin should not take $x as the JoinType with building left side")
+          throw QueryExecutionErrors.hashJoinCannotTakeJoinTypeWithBuildLeftError(x)
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala:73]: this is also assert-like.

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala:73]: ditto to other changes in this file

Comment (beliefer) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala:73]: OK

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by beliefer** [comment]
@cloud-fan Thanks for your work. @allisonwang-db @wangyum Thanks for your review.

**Pull request closed without merging**